tests/unit/FastSearchTest.ts

```diff
@@ -0,0 +1,118 @@
+            {
+                data: ['erica', 'banana'],
+                toSearchableString: (data) => data,
+            },
+            {
+                data: ['banana', 'test'],
+                toSearchableString: (data) => data,
+            },
+        ]);
+
+        expect(search('es')).toEqual([[], ['test']]);
+    });
+
+    it('should work with numbers', () => {
+        const {search} = FastSearch.createFastSearch([
+            {
+                data: [1, 2, 3, 4, 5],
+                toSearchableString: (data) => String(data),
+            },
+        ]);
+
+        expect(search('2')).toEqual([[2]]);
+    });
+
+    it('should work with unicodes', () => {
+        const {search} = FastSearch.createFastSearch([
+            {
+                data: ['banana', 'ñèşťǒř', 'test'],
+                toSearchableString: (data) => data,
+            },
+        ]);
+
+        expect(search('èşť')).toEqual([['ñèşťǒř']]);
+    });
+
+    it('should work with words containing "reserved special characters"', () => {
+        const {search} = FastSearch.createFastSearch([
+            {
+                data: ['ba|nana', 'te{st', 'he}llo'],
+                toSearchableString: (data) => data,
+            },
+        ]);
+
+        expect(search('st')).toEqual([['te{st']]);
+        expect(search('llo')).toEqual([['he}llo']]);
+        expect(search('nana')).toEqual([['ba|nana']]);
+    });
+
+    it('should be case insensitive', () => {
+        const {search} = FastSearch.createFastSearch([
+            {
+                data: ['banana', 'TeSt', 'TEST', 'X'],
+                toSearchableString: (data) => data,
+            },
+        ]);
+
+        expect(search('test')).toEqual([['TeSt', 'TEST']]);
+    });
+
+    it('should work with large random data sets', () => {
+        const data = Array.from({length: 1000}, () => {
+            return Array.from({length: Math.floor(Math.random() * 22 + 9)}, () => {
+                const alphabet = 'abcdefghijklmnopqrstuvwxyz0123456789@-_.';
+                return alphabet.charAt(Math.floor(Math.random() * alphabet.length));
+            }).join('');
+        });
+
+        const {search} = FastSearch.createFastSearch([
+            {
+                data,
+                toSearchableString: (x) => x,
+            },
+        ]);
+
+        data.forEach((word) => {
+            expect(search(word)).toEqual([expect.arrayContaining([word])]);
+        });
+    });
+
+    it('should find email addresses without dots', () => {
+        const {search} = FastSearch.createFastSearch([
+            {
+                data: ['test.user@example.com', 'unrelated'],
+                toSearchableString: (data) => data,
+            },
+        ]);
+
+        expect(search('testuser')).toEqual([['test.user@example.com']]);
+        expect(search('test.user')).toEqual([['test.user@example.com']]);
+        expect(search('examplecom')).toEqual([['test.user@example.com']]);
+    });
+});
```

Comment:
These tests are good. Should we test some common formats of reports and personal details? Feels like there are a few recurring patterns that could be easily added here.

Reply:
Ok. Sounds like the benefit of doing it now is that the tests would help catch potential issues?  I don't think it matters as much as long as all the mainline cases for searching are working (which they seem to be). Let's maybe just plan on doing that follow up plan with some urgency.